### PR TITLE
feat: [FFM-1782]: Added serve default ON variation dropdown/card

### PR DESCRIPTION
**Pull request by chrisblakely01**
##### Summary:

Adds the new "Serve default rule when flag enabled" section to the new targeting tab

Note: This is hidden behind feature flag and not visible to end users as of yet

##### Jira Links:

https://harness.atlassian.net/browse/FFM-1782

##### Coverage for new components:
Screenshot 2022-03-03 at 17 33 57

##### Screenshots:

Screenshot 2022-03-03 at 16 30 04

##### Video:

https://user-images.githubusercontent.com/10871707/156608231-de9b10dd-9c4b-4d07-9772-c2e19c3588ea.mov

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Review by crystal-fong — APPROVED** [review]

File: src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/TargetingRulesTab.tsx [hunk 1]
```diff
@@ -30,7 +33,10 @@ const TargetingRulesTab = ({
   refetchFlagLoading
 }: TargetingRulesTabProps): ReactElement => {
   const initialValues = {
-    state: featureFlagData.envProperties?.state as string
+    state: featureFlagData.envProperties?.state as string,
+    onVariation: featureFlagData.envProperties?.defaultServe.variation
+      ? featureFlagData.envProperties?.defaultServe.variation
```
Comment (willd0yle) [inline src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/TargetingRulesTab.tsx:38]: can `defaultServe` be `undefined`?

could you get `cannot read defaultServe of undefined` error?

**Review by willd0yle — APPROVED** [review]
Just one question but apart from that it looks good

File: src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/TargetingRulesTab.tsx (hunk 1 above)
Comment (chrisblakely01) [reply src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/TargetingRulesTab.tsx:38]: Good question - if envProperties exist, the default serve has to exist. Its basically setting flag rules for different environments.

Comment (chrisblakely01) [reply src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/TargetingRulesTab.tsx:38]: This is typed as well and generated via the API swagger so its in sync with that

**Pull request merged**